### PR TITLE
nushell: update to reflect new settings style

### PR DESCRIPTION
**Pull request by schuelermine**
### Description

Updated the nushell module to reflect the current nushell configuration style.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: tests/modules/services/window-managers/sway/sway-no-xwayland.nix [hunk 1]
```diff
@@ -17,7 +17,7 @@
         ${
           pkgs.writeText "expected" ''
             xwayland disable
-                    ''
```
Comment (sumnerevans) [inline tests/modules/services/window-managers/sway/sway-no-xwayland.nix:20]: please remove this change. It seems entirely unrelated

**Review by sumnerevans — CHANGES_REQUESTED** [review]

File: modules/programs/nushell.nix [hunk 2]
```diff
@@ -21,7 +29,27 @@ in {
       description = "The package to use for nushell.";
     };
 
-    settings = mkOption {
+    config = mkOption {
+      type = types.lines;
+      default = "";
+      defaultText = literalExpression ''""'';
```
Comment (sumnerevans) [inline modules/programs/nushell.nix:35]: I don't think `defaultText` isn't necessary for default-empty strings (at least I can't find another instance of that in the repo.

**Comment by rycee** [comment]
Seems like a duplicate of #2957.

**Comment by rycee** [comment]
I believe this is fixed since the merge of the PR mentioned previously.

**Pull request closed without merging**